### PR TITLE
feat(eth-staking): add unstake pending state, pool stats, stake data …

### PR DESCRIPTION
**Pull request by StanCheb**
## Description

The pending unstake state is added to the staking tab dashboard.
The progress bar is updated accordingly to the pending unstake state.
All fiat related data is hidden when rates are unavailable.
Stake, unstake, claim balances are fetched from Blockbook.
StakingCard is refactored.
Limits validation in the stake and unstake forms is fixed. Now it operates with BigNumber instead of native JS numbers. This caused decimals and scientific notation problems.
Evestake pool stats are added (APY + Next reward payout).
The successful claim notification is added.

## Related Issue

Resolve 
https://github.com/trezor/trezor-suite/issues/9280
https://github.com/trezor/trezor-suite/issues/9246

## Screenshots:
Знімок екрана 2024-02-03 о 12 53 50
Знімок екрана 2024-02-03 о 12 56 04
Знімок екрана 2024-02-03 о 12 57 14
Знімок екрана 2024-02-03 о 12 57 23

### CODE REVIEW

File: suite-common/wallet-utils/src/accountUtils.ts [hunk 1]
```diff
@@ -582,6 +583,35 @@ export const isTestnet = (symbol: NetworkSymbol) => {
     return net?.testnet ?? false;
 };
 
+const areStakingPoolsEqual = (a1: StakingPool[] | undefined, a2: StakingPool[] | undefined) => {
```
Comment (tomasklim) [inline suite-common/wallet-utils/src/accountUtils.ts]: Do you think that comparison like

```
    return JSON.stringify(a1) === JSON.stringify(a2);
``` 
is not enough?

File: packages/suite/src/views/wallet/staking/components/EthStakingDashboard/hooks/useEverstakePoolStats.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            ethApy: BACKUP_ETH_APY,
+            nextRewardPayout: null,
+        },
+    );
+    const [isPoolStatsLoading, setIsPoolStatsLoading] = useState(false);
+
+    useEffect(() => {
+        const abortController = new AbortController();
+
+        const getEeverstakePoolStats = async () => {
```
Comment (tomasklim) [inline packages/suite/src/views/wallet/staking/components/EthStakingDashboard/hooks/useEverstakePoolStats.ts]:
```suggestion
        const getEverstakePoolStats = async () => {
```

File: packages/suite/src/views/wallet/staking/components/EthStakingDashboard/hooks/useEverstakePoolStats.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from 'react';
+import BigNumber from 'bignumber.js';
+import { BACKUP_ETH_APY } from 'src/constants/suite/ethStaking';
+
+export const useEverstakePoolStats = () => {
```
Comment (tomasklim) [inline packages/suite/src/views/wallet/staking/components/EthStakingDashboard/hooks/useEverstakePoolStats.ts:5]: You forgot to use this hook in `StakeEthCard` and `EmptyStakingCard` for live `apy`

File: packages/suite/src/components/wallet/WalletLayout/AccountBanners/StakeEthBanner.tsx [hunk 4]
```diff
@@ -81,7 +81,7 @@ export const StakeEthBanner = () => {
                             <Translation id="TR_STAKE_ETH_EARN_REPEAT" />
                         </Title>
                         <P weight="medium">
-                            <Translation id="TR_STAKE_ANY_AMOUNT_ETH" />
+                            <Translation id="TR_STAKE_ANY_AMOUNT_ETH" values={{ apyPercent: 5 }} />
```
Comment (tomasklim) [inline packages/suite/src/components/wallet/WalletLayout/AccountBanners/StakeEthBanner.tsx:84]: data from `useEverstakePoolStats`?

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/ClaimModal/ClaimEthForm.tsx [hunk 5]
```diff
@@ -105,7 +101,7 @@ export const ClaimEthForm = () => {
                     <Translation id="TR_STAKE_CLAIMING_PERIOD" />
                 </GreyP>
 
-                <div>{claimingPeriod}</div>
+                <Translation id="TR_STAKE_IN_NEXT_BLOCK" />
```
Comment (tomasklim) [inline packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/ClaimModal/ClaimEthForm.tsx]: TR_STAKE_CLAIM_IN_NEXT_BLOCK

**Review by tomasklim — CHANGES_REQUESTED** [review]
I was able to claim when claim button was not available in UI anymore
1. 0.12345 THOL was ready to claim
2. I unstaked 0.1 THOL
3. Before the transaction was confirmed, I opened claim modal
4. When it was confirmed, I claimed the balance
5. Success. Is it ok?

Let's say, that I stake 1 THOL, I unstake 0.95. 0.05 THOL is there, which I cannot unstake because of 0.1 limit
Screenshot 2024-02-04 at 13 38 39

Also when you check the image above, when Stake, rewards, and unstaking does not fit, it would be nice to put them under each other

File: suite-common/wallet-utils/src/accountUtils.ts (hunk 1 above)
Comment (tomasklim) [reply suite-common/wallet-utils/src/accountUtils.ts]: If not, just please use different name for variables `a1` and `a2`

**Comment by omykhailovskyi** [comment]
> I was able to claim when claim button was not available in UI anymore
> 
> 1. 0.12345 THOL was ready to claim
> 2. I unstaked 0.1 THOL
> 3. Before the transaction was confirmed, I opened claim modal
> 4. When it was confirmed, I claimed the balance
> 5. Success. Is it ok?
> 
> Let's say, that I stake 1 THOL, I unstake 0.95. 0.05 THOL is there, which I cannot unstake because of 0.1 limit Screenshot 2024-02-04 at 13 38 39
> 
> Also when you check the image above, when Stake, rewards, and unstaking does not fit, it would be nice to put them under each other

Hey, @tomasklim 
Answering your question, think it's fine, but we can discuss in voice.
So basically we at the moment provide possibility to claim ready to claim balance, but to avoid user to spend more on tx fee, he claims all amount.
For the case if user already has ready to claim balance we represent it on the Unstake Modal form
image
Please check if it's fine.

Currently we have a min unstake amount and it's the same as min stake amount (0.1 ETH) to minimize spam activities.

File: packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/ClaimModal/ClaimEthForm.tsx (hunk 5 above)
Comment (StanCheb) [reply packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/ClaimModal/ClaimEthForm.tsx]: https://github.com/everstake/trezor-suite/pull/10/commits/81f13c29cc0a25435fad0c613489f4a7942b1dcb

File: packages/suite/src/components/wallet/WalletLayout/AccountBanners/StakeEthBanner.tsx (hunk 4 above)
Comment (StanCheb) [reply packages/suite/src/components/wallet/WalletLayout/AccountBanners/StakeEthBanner.tsx:84]: I actually thought about it, but then I saw the text with the "up to" wording there and thought that it should be static data. Should I use the dynamic value anyway?

File: packages/suite/src/views/wallet/staking/components/EthStakingDashboard/hooks/useEverstakePoolStats.ts (hunk 3 above)
Comment (StanCheb) [reply packages/suite/src/views/wallet/staking/components/EthStakingDashboard/hooks/useEverstakePoolStats.ts:5]: I actually thought about it, but then I saw the text with the "up to" wording there and thought that it should be static data. Should I use the dynamic value anyway?

File: packages/suite/src/views/wallet/staking/components/EthStakingDashboard/hooks/useEverstakePoolStats.ts (hunk 2 above)
Comment (StanCheb) [reply packages/suite/src/views/wallet/staking/components/EthStakingDashboard/hooks/useEverstakePoolStats.ts]: https://github.com/everstake/trezor-suite/pull/10/commits/81f13c29cc0a25435fad0c613489f4a7942b1dcb

File: suite-common/wallet-utils/src/accountUtils.ts (hunk 1 above)
Comment (StanCheb) [reply suite-common/wallet-utils/src/accountUtils.ts]: If the order of the fields inside `stakingPools` objects is guaranteed and data there will always be of a `string` type, then it should be enough, yes.

https://github.com/everstake/trezor-suite/pull/10/commits/81f13c29cc0a25435fad0c613489f4a7942b1dcb

**Comment by omykhailovskyi** [comment]
@shotgunofdeath Move Claim message in the top before Only rewards line

File: packages/suite/src/views/wallet/staking/components/EthStakingDashboard/hooks/useEverstakePoolStats.ts (hunk 3 above)
Comment (tomasklim) [reply packages/suite/src/views/wallet/staking/components/EthStakingDashboard/hooks/useEverstakePoolStats.ts:5]: Let's see the result here https://satoshilabs.slack.com/archives/C0543DJBK0C/p1707311776921939

**Review by tomasklim — APPROVED** [review]

**Pull request merged**